### DISCUSSION TITLE
LayoutOptions ...AndExpand obsolete, ok to use new LayoutOptions(LayoutAlignment...., expand: true);

### DISCUSSION THREAD

**Discussion by 3sRykaert** [dotnet/maui#20774] [Q&A] [answered]
Since (for example) `LayoutOptions.StartAndExpand` is obsolete (and probably will be removed in future version) is it ok to just replace it by: `new LayoutOptions(LayoutAlignment.Start, true)`

Will this still work in future?
Since the obsolete warning, tells to use a grid instead, see:
https://github.com/dotnet/maui/blob/85adc13a249fe4ac59da6518ea5dc70055ee569c/src/Controls/src/Core/LayoutOptions.cs#L22

**Comment by 3sRykaert** [comment]
@jfversluis or @hartez (since you guys show up in the history of [maui/src/Controls/src/Core/LayoutOptions.cs](https://github.com/dotnet/maui/blob/85adc13a249fe4ac59da6518ea5dc70055ee569c/src/Controls/src/Core/LayoutOptions.cs#L22)) you can answer this by accident?

**Reply by jfversluis** [reply, marked as answer]
Maybe something in here will be helpful for you: https://github.com/dotnet/maui/blob/main/docs/design/layout.md

**Reply by 3sRykaert** [reply]
Thanks, this indeed seems a clear answer:
`For ease of migration, Controls still uses the LayoutOptions structure, which also includes the (deprecated) "AndExpand" flags. The Core layout system ignores the "AndExpand" flags and translates the Fill/Start/Center/End values from LayoutOptions to the LayoutAlignment enumeration.`
`This brings us to the second main difference: the MAUI VerticalStackLayout and HorizontalStackLayout do not recognize the "AndExpand" layout options. If they see a child with layout options that include "AndExpand", they simply treat it as if the "AndExpand" weren't there - e.g., FillAndExpand becomes Fill.`
